cmd/webhook/main.go

```diff
@@ -19,15 +19,23 @@ package main
 import (
 	"os"
 
+	"github.com/go-logr/logr"
 	"k8s.io/component-base/cli"
 	_ "k8s.io/component-base/logs/json/register" // for JSON log format registration
 	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	"github.com/karmada-io/karmada/cmd/webhook/app"
 )
 
 func main() {
 	ctx := controllerruntime.SetupSignalHandler()
+	// Controller-runtime expects its consumers to set a logger through log.SetLogger.
+	// If SetLogger is not called within the first 30 seconds of a binaries lifetime, it will get
+	// set to a NullLogSink and report an error. Here's to silence the "log.SetLogger(...) was never called; logs will not be displayed" error
+	// by setting a logger through log.SetLogger.
+	// More info refer to: https://github.com/karmada-io/karmada/issues/4868.
+	controllerruntime.SetLogger(logr.New(log.NullLogSink{}))
```

Comment:
https://github.com/kubernetes-sigs/cluster-api/blob/edc8348877aa08f15de24564f0f1dd121b408f5b/main.go#L245

just a question,  can be use `klog.Background()`  ?

Reply:
I'll give it a try and get back with feedback

Reply:
> just a question, can be use `klog.Background()` ?

Yes, even better！